src/cpp/src/visual_language/qwen2vl/classes.cpp

```diff
@@ -397,53 +510,82 @@ ov::Tensor get_attention_mask(const std::vector<std::array<size_t, 3>>& reordere
     return t_cu_seqlens;
 }
 
-ov::Tensor concatenate_image_embeds(const std::vector<ov::Tensor>& reordered_image_embeds) {
+ov::Tensor concatenate_video_image_embeds(const std::vector<ov::Tensor>& reordered_video_embeds, const std::vector<ov::Tensor>& reordered_image_embeds) {
+    // one image + zero video.
+    if (reordered_image_embeds.size() == 1u && reordered_video_embeds.empty()) {
+        return reordered_image_embeds.at(0);
+    }
+    // zero image + one video.
+    if (reordered_image_embeds.empty() && reordered_video_embeds.size() == 1u) {
+        return reordered_video_embeds.at(0);
+    }
+    // zero image + zero video.
+    if (reordered_image_embeds.empty() && reordered_video_embeds.empty()) {
+        return ov::Tensor();
+    }
+
+    // mulitple image(s) or video(s).
     ov::Tensor concatenated_embeds;
-    if (reordered_image_embeds.size() == 1) {
-        concatenated_embeds = reordered_image_embeds.at(0);
-    } else {
-        size_t total_length = 0;
-        for (const auto& embed : reordered_image_embeds) {
-            total_length += embed.get_shape().at(0);
-        }
-        size_t hidden_dim = reordered_image_embeds.at(0).get_shape().at(1);
-        
-        concatenated_embeds = ov::Tensor(reordered_image_embeds.at(0).get_element_type(), {total_length, hidden_dim});
-        float* concat_data = concatenated_embeds.data<float>();
-        
-        size_t offset = 0;
-        for (const auto& embed : reordered_image_embeds) {
-            size_t embed_size = embed.get_shape().at(0) * embed.get_shape().at(1);
-            std::memcpy(concat_data + offset, embed.data(), embed.get_byte_size());
-            offset += embed_size;
-        }
+    size_t total_length = 0;
+    for (const auto& embed : reordered_video_embeds) {
+        total_length += embed.get_shape().at(0);
+    }
+    for (const auto& embed : reordered_image_embeds) {
+        total_length += embed.get_shape().at(0);
+    }
+
+    // The video and image embeds features are from same embeded model.
+    // So reordered_image_embeds and reordered_video_embeds should have same element type and hidden_dim.
+    if (reordered_video_embeds.size() > 0u && reordered_image_embeds.size() > 0u) {
+        OPENVINO_ASSERT(reordered_video_embeds.at(0).get_element_type() == reordered_image_embeds.at(0).get_element_type());
+        OPENVINO_ASSERT(reordered_video_embeds.at(0).get_shape().at(1) == reordered_image_embeds.at(0).get_shape().at(1));
+    }
+
+    size_t hidden_dim;
+    ov::element::Type type;
+    if (reordered_image_embeds.size() > 0u) {
+        hidden_dim = reordered_image_embeds.at(0).get_shape().at(1);
+        type = reordered_image_embeds.at(0).get_element_type();
+    } else if (reordered_video_embeds.size() > 0u) {
+        hidden_dim = reordered_video_embeds.at(0).get_shape().at(1);
+        type = reordered_video_embeds.at(0).get_element_type();
+    }
```

Comment:
you don't need this `if-else` to get `hidden_dim` and `type` because you have already checked sizes of `reordered_image_embeds` and `reordered_video_embds` above. See:

<img width="723" height="245" alt="image" src="https://github.com/user-attachments/assets/1add3bbd-7434-4e91-a20e-9a140740e782" />